SonarScanner.Shim/PropertiesFileGenerator.cs

```diff
@@ -327,16 +370,29 @@ private static string TryGetCodeCoverageReport(ProjectInfo project, ILogger logg
         }
 
         /// <summary>
-        /// Returns all of the analysis properties that should
-        /// be written to the sonar-project properties file
+        /// Returns the analysis properties specified through the call.
         /// </summary>
-        private static AnalysisProperties GetAnalysisPropertiesToWrite(AnalysisConfig config, ILogger logger)
+        private static AnalysisProperties GetAnalysisProperties(AnalysisConfig config)
         {
             AnalysisProperties properties = new AnalysisProperties();
 
             properties.AddRange(config.GetAnalysisSettings(false).GetAllProperties()
-                // Strip out any sensitive properties
-                .Where(p => !p.ContainsSensitiveData()));
+                      // Strip out any sensitive properties
+                      .Where(p => !p.ContainsSensitiveData()));
+
+            return properties;
+        }
+
+        /// <summary>
+        /// Returns all of the analysis properties that should be written to the sonar-project properties file.
+        /// </summary>
+        private static AnalysisProperties GetAnalysisPropertiesToWrite(AnalysisProperties properties, ILogger logger)
+        {
+            Property encodingProperty;
+            if (Property.TryGetProperty(SonarProperties.SourceEncoding, properties, out encodingProperty))
+            {
+                properties.Remove(encodingProperty);
```

Comment:
Not sure there's an easy way to improve, but a method named `Get*` with the side effect of mutating its parameter is usually a code smell.